src/middlewared/middlewared/plugins/nvmet/kernel.py

```diff
@@ -0,0 +1,782 @@
+    def src_dir_name(cls, entry, render_ctx: dict):
+        return f'{entry[cls.src_query_keys[0]][cls.src_query_keys[1]]}'
+
+    @classmethod
+    def dst_name(cls, entry):
+        return f'{entry[cls.dst_query_keys[0]][cls.dst_query_keys[1]]}'
+
+
+class NvmetPortSubsysConfig(NvmetLinkConfig):
+    query = 'nvmet.port_subsys.query'
+    src_parentdir = 'ports'
+    src_subdir = 'subsystems'
+    src_query_keys = ['port', 'nvmet_port_index']
+    dst_dir = 'subsystems'
+    dst_query_keys = ['subsys', 'nvmet_subsys_subnqn']
+
+    @classmethod
+    def src_dir_name(cls, entry, render_ctx: dict):
+        # Because we have elected to support overriding the global ANA
+        # setting for individual subsystems this has two knock-on effects
+        # 1. Additional ANA-specific port indexes are in injected
+        # 2. Particular subsystems will link to either the ANA or non-ANA
+        #    port index.
+        # However, if we're on the standby node we never want to setup
+        # a link to the VIP port.
+        raw_index = entry[cls.src_query_keys[0]][cls.src_query_keys[1]]
+        # Now check whether ANA is playing a part.
+        match entry['subsys']['nvmet_subsys_ana']:
+            case True:
+                index = raw_index + ANA_PORT_INDEX_OFFSET
+            case False:
+                index = raw_index
+            case _:
+                if render_ctx['nvmet.global.ana_enabled']:
+                    index = raw_index + ANA_PORT_INDEX_OFFSET
+                else:
+                    index = raw_index
+
+        if index < ANA_PORT_INDEX_OFFSET and render_ctx['failover.status'] == 'BACKUP':
+            return None
+
+        return str(index)
+
+    @classmethod
+    def dst_name(cls, entry):
+        return f'{entry[cls.dst_query_keys[0]][cls.dst_query_keys[1]]}'
+
+    @classmethod
+    def create_links(cls, entry: dict):
+        return entry['port']['nvmet_port_enabled']
+
+
+def write_config(config):
+    if not pathlib.Path(NVMET_KERNEL_CONFIG_DIR).exists():
+        return
+
+    # Render operations are context managers that do
+    # 1. Create-style operations
+    # 2. yield
+    # 3. Delete-style operations
+    #
+    # Therefore we can nest them to enfore the necessary
+    # order of operations.
+    with NvmetSubsysConfig.render(config):
+        with NvmetHostConfig.render(config):
+            with NvmetPortConfig.render(config):
+                with NvmetPortReferralConfig.render(config):
+                    with NvmetPortAnaReferralConfig.render(config):
+                        with NvmetHostSubsysConfig.render(config):
+                            with NvmetPortSubsysConfig.render(config):
+                                with NvmetNamespaceConfig.render(config):
+                                    pass
+
+
+def _map_device_path(device_type, device_path):
+    match device_type:
+        case 'FILE':
+            if device_path.startswith('/mnt/'):
+                return device_path
```

Comment:
More fun with case statements.   The `if` part can be included in the `case` statement.
```suggestion
        case 'FILE' if device_path.startswith('/mnt/'):       
                return device_path
```
See https://docs.python.org/3.11/reference/compound_stmts.html#overview

Reply:
No planning to make this change for now.